src/ansys/rocky/core/launcher.py

```diff
@@ -150,6 +150,122 @@ def launch_rocky(
     return client
 
 
+def launch_freeflow(
+    freeflow_exe: Optional[Union[Path, str]] = None,
+    freeflow_version: Optional[int] = None,
+    *,
+    headless: bool = True,
+    server_port: int = DEFAULT_SERVER_PORT,
+    close_existing: bool = False,
+) -> RockyClient:
+    """
+    Launch the Freeflow executable with the PyRocky server enabled.
+
+    This method waits for Rocky to start up and then returns a
+    ```RockyClient`` instance.
+
+    Parameters
+    ----------
+    freeflow_exe:
+        Path to the Freeflow executable.
+    freeflow_version:
+        Freeflow version to run. If no executable is passed and the version is not
+        specified, this method tries to find the path using the latest Ansys path
+        returned by ansys-tools-path API.
+    headless:
+        Whether to launch Freeflow in headless mode. The default is ``True``.
+    server_port:
+        Set the port for Rocky RPC server.
+    close_existing:
+        Checks if a session exists under the given server_port and closes it
+        before attempting to launch a new session.
+
+    Returns
+    -------
+    RockyClient
+        Rocky client instance connected to the launched Rocky/Freeflow app.
+    """
+    if _is_port_busy(server_port):
+        if close_existing:
+            # Will try to connect to an existing session using the
+            # given server port and attempt to close it.
+            client = connect(port=server_port)
+            try:
+                client.close()
+            except CommunicationError:
+                # Maybe the session closed in the meantime so we just pass
+                pass
+        else:
+            raise FreeflowLaunchError(f"Port {server_port} is already in use.")
+
+    ansys_installations = get_available_ansys_installations()
```

Comment:
Did you just duplicated all this code???

Please, move all this code that finds the executable to a unique generic function and use it on both `launch_rocky` and `launch_freeflow`.